indexing-service/src/main/java/org/apache/druid/indexing/common/IndexTaskClient.java

```diff
@@ -387,6 +387,65 @@ private FullResponseHolder submitRequest(
     }
   }
 
+  private FullResponseHolder submitRequest(Request request) throws IOException, ChannelException
+  {
+    try {
+      log.debug("HTTP %s: %s", request.getMethod().getName(), request.getUrl().toString());
+      return httpClient.go(request, new FullResponseHandler(StandardCharsets.UTF_8), httpTimeout).get();
+    }
+    catch (Exception e) {
+      // Check it's IOException, ChannelException or InterruptedException
```

Comment:
Hmm, would it be better to push all of this logic into `throwIfPossible(e);` and make it a void method? I think it might be cleaner and make the handling logic a bit easier to follow since it wouldn't either return a throwable or throw it.

Reply:
Hmm, yeah I see now it can't be void, my bad. I think still having all of the logic inside of `throwIfPossible` might be best, e.g. maybe

```
catch (Exception e) {
  return throwIfPossible(e);
}
```

Reply:
Ok, I did some fix. Please take a look again.